test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -716,18 +750,31 @@ TEST_P(WasmHttpFilterTest, AsyncCall) {
   Http::MockAsyncClientRequest request(&cluster_manager_.thread_local_cluster_.async_client_);
   Http::AsyncClient::Callbacks* callbacks = nullptr;
   cluster_manager_.initializeThreadLocalClusters({"cluster"});
+
+  setupSpan();
+  Tracing::SpanPtr current_span;
+  Http::MockStreamDecoderFilterCallbacks decoder_callbacks;
+  rootContext().setDecoderFilterCallbacks(decoder_callbacks);
+  EXPECT_CALL(decoder_callbacks, activeSpan()).WillOnce(ReturnRef(*datadog_span_));
+
   EXPECT_CALL(cluster_manager_.thread_local_cluster_, httpAsyncClient());
   EXPECT_CALL(cluster_manager_.thread_local_cluster_.async_client_, send_(_, _, _))
-      .WillOnce(
-          Invoke([&](Http::RequestMessagePtr& message, Http::AsyncClient::Callbacks& cb,
-                     const Http::AsyncClient::RequestOptions&) -> Http::AsyncClient::Request* {
+      .WillOnce(Invoke(
+          [&](Http::RequestMessagePtr& message, Http::AsyncClient::Callbacks& cb,
+              const Http::AsyncClient::RequestOptions& options) -> Http::AsyncClient::Request* {
             EXPECT_EQ((Http::TestRequestHeaderMapImpl{{":method", "POST"},
                                                       {":path", "/"},
                                                       {":authority", "foo"},
                                                       {"content-length", "11"}}),
                       message->headers());
             EXPECT_EQ((Http::TestRequestTrailerMapImpl{{"trail", "cow"}}), *message->trailers());
             callbacks = &cb;
+            current_span =
+                options.parent_span_->spawnChild(Tracing::EgressConfig::get(), "fake", start_time_);
+            current_span->injectContext(message->headers(), nullptr);
+            EXPECT_TRUE(message->headers().getByKey("x-datadog-parent-id") != "");
+            EXPECT_TRUE(message->headers().getByKey("x-datadog-parent-id") != "");
```

Comment:
Should one of those be `x-datadog-trace-id` instead?

Reply:
thanks, i will fix this